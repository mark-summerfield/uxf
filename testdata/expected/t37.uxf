uxf 1 MyApp Data
#<Suitable for SQLite conversion>
=Customers CID:int Company:str Address:str Contact:str Email:str
=Invoices INUM:int CID:int Raised_Date:date Due_Date:date Paid:bool Description:str
=Items IID:int INUM:int Delivery_Date:date Unit_Price:real Quantity:int Description:str
[#<There is a 1:M relationship between the Invoices and Items tables>
  (Customers
    50 <Best People> <123 Somewhere> <John Doe> <j@doe.com>
    19 <Supersuppliers> ? <Jane Doe> <jane@super.com>
  )
  (Invoices
    152 50 2022-01-17 2022-02-17 no <COD>
    153 19 2022-01-19 2022-02-19 yes <>
  )
  (Items
    1839 152 2022-01-16 29.99 2 <Bales of hay>
    1840 152 2022-01-16 5.98 3 <Straps>
    1620 153 2022-01-19 11.5 1 <Washers (1-in)>
  )
]
