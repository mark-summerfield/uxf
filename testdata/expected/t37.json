{
  "UXF^custom": "MyApp Data",
  "UXF^comment": "Suitable for SQLite conversion",
  "UXF^ttypes": [
    {
      "UXF^ttype": {
        "name": "Customers",
        "fields": {
          "CID": "int",
          "Company": "str",
          "Address": "str",
          "Contact": "str",
          "Email": "str"
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "Invoices",
        "fields": {
          "INUM": "int",
          "CID": "int",
          "Raised_Date": "date",
          "Due_Date": "date",
          "Paid": "bool",
          "Description": "str"
        }
      }
    },
    {
      "UXF^ttype": {
        "name": "Items",
        "fields": {
          "IID": "int",
          "INUM": "int",
          "Delivery_Date": "date",
          "Unit_Price": "real",
          "Quantity": "int",
          "Description": "str"
        }
      }
    }
  ],
  "UXF^data": {
    "UXF^list": {
      "comment": "There is a 1:M relationship between the Invoices and Items tables",
      "vtype": null,
      "list": [
        {
          "UXF^table": {
            "name": "Customers",
            "comment": null,
            "fields": {
              "CID": "int",
              "Company": "str",
              "Address": "str",
              "Contact": "str",
              "Email": "str"
            },
            "records": [
              [
                50,
                "Best People",
                "123 Somewhere",
                "John Doe",
                "j@doe.com"
              ],
              [
                19,
                "Supersuppliers",
                null,
                "Jane Doe",
                "jane@super.com"
              ]
            ]
          }
        },
        {
          "UXF^table": {
            "name": "Invoices",
            "comment": null,
            "fields": {
              "INUM": "int",
              "CID": "int",
              "Raised_Date": "date",
              "Due_Date": "date",
              "Paid": "bool",
              "Description": "str"
            },
            "records": [
              [
                152,
                50,
                {
                  "UXF^date": "2022-01-17"
                },
                {
                  "UXF^date": "2022-02-17"
                },
                false,
                "COD"
              ],
              [
                153,
                19,
                {
                  "UXF^date": "2022-01-19"
                },
                {
                  "UXF^date": "2022-02-19"
                },
                true,
                ""
              ]
            ]
          }
        },
        {
          "UXF^table": {
            "name": "Items",
            "comment": null,
            "fields": {
              "IID": "int",
              "INUM": "int",
              "Delivery_Date": "date",
              "Unit_Price": "real",
              "Quantity": "int",
              "Description": "str"
            },
            "records": [
              [
                1839,
                152,
                {
                  "UXF^date": "2022-01-16"
                },
                29.99,
                2,
                "Bales of hay"
              ],
              [
                1840,
                152,
                {
                  "UXF^date": "2022-01-16"
                },
                5.98,
                3,
                "Straps"
              ],
              [
                1620,
                153,
                {
                  "UXF^date": "2022-01-19"
                },
                11.5,
                1,
                "Washers (1-in)"
              ]
            ]
          }
        }
      ]
    }
  }
}